uberfire-workbench/uberfire-workbench-client-views-patternfly/src/main/java/org/uberfire/client/views/pfly/notifications/NotificationPopupsManagerView.java

```diff
@@ -77,14 +84,23 @@ public NotificationPopupHandle show(NotificationEvent event,
         view.setNotificationWidth(getWidth() + "px");
 
         if (event.hasNavigation()) {
-            view.addNavigation(event.getNavigationText(), () -> {
-                hideCommand.execute();
-                placeManager.goTo(event.getNavigationPath());
-            });
+            ResourceRef resourceRef = new ResourceRef(
```

Comment:
I think a better approach to keep this permission check more flexible is to receive a `Promise<Boolean>` in the `setNavigation` method, and check it here. Mainly because there could be a case where the navigation needs additional checks to be added or not, or isn't even a perspective.